src/photos/ducks/timeline-clusters/index.js

```diff
@@ -36,39 +38,122 @@ const TIMELINE_MUTATIONS = query => ({
     })
 })
 
-const getPhotosByMonth = photos => {
-  let sections = {}
-  photos.forEach(p => {
+const formatDate = date => {
+  return format(date, 'DD MMMM YYYY')
+}
+
+const sectionTitle = album => {
+  if (album.period) {
+    const startPeriod = new Date(album.period.start)
+    const endPeriod = new Date(album.period.end)
+    if (differenceInCalendarDays(endPeriod, startPeriod) > 0) {
+      // TODO Better period display
+      return formatDate(startPeriod) + ' - ' + formatDate(endPeriod)
+    }
+  }
+  return formatDate(new Date(album.name))
+}
+
+/**
+ * A section matches if:
+ * - The datetime is inside the period.
+ * - The section has no period (not a cluster) and is the datetime's day.
+ */
+const matchingSection = (sections, datetime) => {
+  return Object.keys(sections).find(date => {
+    if (sections[date].period) {
+      const startPeriod = new Date(sections[date].period.start)
+      const endPeriod = new Date(sections[date].period.end)
+      return datetime >= startPeriod && datetime <= endPeriod
+    } else {
+      // If the section has no period, it is not a cluster but a daily section
+      return differenceInCalendarDays(datetime, new Date(date)) === 0
+    }
+  })
+}
+
+/**
+ * Save the photos not clustered in the sections.
+ * It simulates a clustering, waiting for the actual one to be processed.
+ */
+const handlePhotosNotClustered = (sections, photosNotClustered) => {
+  photosNotClustered.forEach(photo => {
     const datetime =
-      p.metadata && p.metadata.datetime ? p.metadata.datetime : Date.now()
-    // here we want to get an object whose keys are months in a l10able format
-    // so we only keep the year and month part of the date
-    const month = datetime.slice(0, 7) + '-01T00:00'
-    /* istanbul ignore else */
-    if (!sections.hasOwnProperty(month)) {
-      sections[month] = []
+      photo.metadata && photo.metadata.datetime
+        ? photo.metadata.datetime
+        : photo.created_at
+
+    const sectionDate = matchingSection(sections, new Date(datetime))
+    if (sectionDate) {
+      sections[sectionDate].photos.push(photo)
+    } else {
+      // Create a new section for this day, without a period, to differentiate from clusters' sections
+      const day = datetime.slice(0, 10)
```

Comment:
This still looks odd to me, isn't there a `Date` method that formats the datetime as we want it?

Reply:
We could use the [`toLocaleDateString`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Date/toLocaleDateString) but we have to pass a locale. Or we could use the date-fns `format`, but I'm not sure how this would be better as we know the date format here is YYYY-MM-DD

Reply:
Do we really know this though? The data comes from `photo.metadata.datetime` or `photo.created_at` and hasn't been parsed. `created_at` is almost certainly a standard time string, but can `datetime` be something we don't expect?
I also think that with `const day = f(new Date(datetime), 'YYYY-MM-DD')`, it's really obvious what we're getting. With `.slice(0, 10)` the reader first has to figure out what the variable contains.
Another option would be to put this in a named function, — so it becomes something like `const day = getSectionDate(datetime)`?